inc/azure_c_util/thandle.h

```diff
@@ -411,6 +416,34 @@ void THANDLE_MOVE(T)(THANDLE(T) * t1, THANDLE(T) * t2 )
+        (t1 == NULL) ||                                                                                                                                             \
+        /*Codes_SRS_THANDLE_01_002: [ If t2 is NULL then THANDLE_INITIALIZE_MOVE shall return. ]*/                                                                  \
+        (t2 == NULL)                                                                                                                                                \
+    )                                                                                                                                                               \
+    {                                                                                                                                                               \
+        LogError("invalid argument THANDLE(" MU_TOSTRING(T) ") * t1=%p, THANDLE(" MU_TOSTRING(T) ") t2=%p", t1, t2 );                                               \
+    }                                                                                                                                                               \
+    else                                                                                                                                                            \
+    {                                                                                                                                                               \
+        if (*t2 == NULL)                                                                                                                                            \
+        {                                                                                                                                                           \
+            /*Codes_SRS_THANDLE_01_003: [ If *t2 is NULL then THANDLE_INITIALIZE_MOVE shall THANDLE_DEC_REF *t1, set *t1 to NULL and return. ]*/                    \
+            *(T const**)t1 = NULL;                                                                                                                                  \
+        }                                                                                                                                                           \
+        else                                                                                                                                                        \
+        {                                                                                                                                                           \
+            /*Codes_SRS_THANDLE_01_004: [ If *t2 is not NULL then THANDLE_INITIALIZE_MOVE shall set *t1 to *t2, set *t2 to NULL and return. ]*/                     \
```

Comment:
why is there no DEC_REF here?

Reply:
It is a move, the instance moves from t2 to t1. No ref count touches are needed.

---
In reply to: [470884539](https://github.com/Azure/azure-c-util/pull/60#discussion_r470884539) [](ancestors = 470884539)